### DISCUSSION TITLE
How to open master window pre-maximized ?

### DISCUSSION THREAD

**Discussion by sharneil-sh7** [TomSchimansky/CustomTkinter#1211] [Q&A] [answered]
Hi! When I run the code, GUI opens in a window hanging somewhere on the screen. It is required to manually maximize if wanted. I wish to open a window already maximized.

Consider simple code:
> import customtkinter
> master = customtkinter.CTk()
> master.mainloop()

This opens as shown below and then I need to maximize it manually.
![image](https://user-images.githubusercontent.com/124943723/219907548-f469329b-1bcd-4055-9c6e-8736d7f7456d.png)

**What changes should I make in the lines of code above to open the window maximized by default?**

Adding line 'master.geometry("1920x1080-10+0")' between lines 2 and 3 kinda works. But technically still not maximized.
In the earlier version of customtkinter, 'master.state("zoomed")' used to work perfectly, but not anymore. Still works in 'Tkinter' though.

**Comment by sharneil-sh7** [comment]
In the earlier version of customtkinter, 'master.state("zoomed")' used to work perfectly. But not anymore.

**Reply by Wolf-SO** [reply]
> In the earlier version of customtkinter, 'master.state(zoomed)' used to work perfectly. But not anymore.

As far as I know, it should be `master.state("zoomed")`, note a `str` as argument in general. 

See also [tkinter python maximize window - Stack Overflow](https://stackoverflow.com/questions/15981000/tkinter-python-maximize-window "tkinter python maximize window - Stack Overflow")

**Comment by saiyam-sandhir** [comment]
> Hi! When I run the code, GUI opens in a window hanging somewhere on the screen. It is required to manually maximize if wanted. I wish to open a window already maximized.
> 
> Consider simple code:
> 
> > import customtkinter
> > master = customtkinter.CTk()
> > master.mainloop()
> 
> This opens as shown below and then I need to maximize it manually. ![image](https://user-images.githubusercontent.com/124943723/219907548-f469329b-1bcd-4055-9c6e-8736d7f7456d.png)
> 
> **What changes should I make in the lines of code above to open the window maximized by default?**
> 
> Adding line 'master.geometry("1920x1080-10+0")' between lines 2 and 3 kinda works. But technically still not maximized. In the earlier version of customtkinter, 'master.state("zoomed")' used to work perfectly, but not anymore. Still works in 'Tkinter' though.

try using ` root.wm_state("zoomed") `

I had the same problem and it worked for me

**Reply by sharneil-sh7** [reply]
I tried this. But did not work in the newer version.

Found a workaround thanks to @ElectricCandlelight :
_root.after(0, lambda:root.state('zoomed'))_

**Comment by ElectricCandlelight** [comment, marked as answer]
Duplicate of issue #1223 

Try this work around.

```Python
import customtkinter

root = customtkinter.CTk()
root.after(0, lambda:root.state('zoomed'))
root.mainloop()
```

**Reply by sharneil-sh7** [reply]
Thanks! This worked!
